spark/src/main/scala/org/apache/comet/rules/CometScanRule.scala

```diff
@@ -258,11 +258,15 @@ case class CometScanRule(session: SparkSession) extends Rule[SparkPlan] {
   }
 
   private def selectScan(scanExec: FileSourceScanExec, partitionSchema: StructType): String = {
-    // TODO these checks are not yet exhaustive. For example, native_iceberg_compat does
-    //  not support reading from S3
 
     val fallbackReasons = new ListBuffer[String]()
 
+    // native_iceberg_compat only supports local filesystem and S3
+    if (!scanExec.relation.inputFiles
+        .forall(path => path.startsWith("file://") || path.startsWith("s3a://"))) {
```

Comment:
@parthchandra I may need some guidance here. Just looking at the input files causes some stats to be updated (such as `HiveCatalogMetrics.METRIC_FILES_DISCOVERED` and `HiveCatalogMetrics.METRIC_FILE_CACHE_HITS`), leading to some test failures. I wonder if this could be adding some overhead.

Do you know if there is a more efficient way for us to check for supported file locations?

Reply:
Thanks @parthchandra. This is now ready for review.